### ISSUE TITLE
"Waypoint upload not complete" when trying to start mission that is ready to execute

### ISSUE THREAD

**Issue by cmbasnett** [dji-sdk/Mobile-SDK-iOS#224]
Model: Phantom 4 Pro
Firmware: 01.05.0300

I'm trying to start a simulator mission with a simple set of simple waypoints. The waypoints are successfully being uploaded (see log below) and the `DJIWaypointMissionOperator`'s `currentState`is `DJIWaypointMissionStateReadyToExecute`. Unfortunately when I call `startMissionWithCompletion` on the operator, it immediately calls the completion block with an an error stating:

> Waypoint upload not complete.

Below is the set of instructions being sent to the SDK (I'm wrapping parts of the SDK functionality for ReactNative, so it's in Javascript but it maps 1:1 to the API calls).

```
let location = {latitude: 49.106179, longitude: -123.074116}
await DJISimulator.start(location, 20, 10)
try {
    let missionId = await DJIWaypointMission.allocate()
    let coordinates = [...] // truncated for brevity
    for (let i = 0; i < coordinates.length; ++i) {
	await DJIWaypointMission.addWaypoint(missionId, coordinates[i])
    }
    let operatorId = await DJIMissionControl.waypointMissionOperator()
    await DJIWaypointMissionOperator.loadMission(operatorId, missionId)
    await DJIWaypointMissionOperator.uploadMission(operatorId)
    await DJIFlightController.startTakeoff()
    await sleep(10000)
    await DJIFlightController.setHomeLocationUsingCurrentAircraftLocation()
    await DJIWaypointMissionOperator.startMission(operatorId) // error thrown here!
```

Below is the upload progress being reported to a remote logging system. Note that the last entry reports the `currentState` as `5` (`DJIWaypointMissionStateReadyToExecute`):

![screen shot 2018-03-02 at 10 32 07 am](https://user-images.githubusercontent.com/5035660/36915495-cc7d139a-1e05-11e8-87c1-fea65ebde765.png)

And for completeness, here is the actual Objective-C code that calls `startMissionWithCompletion`. Note the `assert`s checking that `currentState` is the expected value:
```
RCT_REMAP_METHOD(startMission,
                   startMission:(NSString*)waypointMissionOperatorId
                   resolver:(RCTPromiseResolveBlock)resolve
                   rejector:(RCTPromiseRejectBlock)reject) {
    DJIWaypointMissionOperator* waypointMissionOperator = [[RCTNativeObjectTable shared] get:waypointMissionOperatorId];
    assert(waypointMissionOperator.currentState == DJIWaypointMissionStateReadyToExecute);
    [waypointMissionOperator startMissionWithCompletion:^(NSError * _Nullable error) {
      if (error != nil) {
        SPX_REJECT_WITH_ERROR(reject, error) // Error is thrown here.
      } else {
        resolve(nil);
      }
    }];
  }
```

Any help is very much appreciated! Thank you!

**Comment by cmbasnett** [comment]
Fixed it. Turns out the problem was fixed by removing the `startTakeoff` call. I'm not sure why exactly this fixed it, I can only assume that calling `startTakeoff` resets an internal state that makes the drone forget the state of the waypoint mission.

I looked through the documentation and I couldn't find anything that would indicate the source of the problem. Perhaps someone from DJI could shed some light on the problem so others can avoid running into the same problem.
